<launch>
  <arg name="simulation" default="$(env SIMULATION)"/>

<!-- SIMULATION -->
  <group if="$(arg simulation)">

  	<!-- Set the body identifier and names that the mocap script need to track -->
  	<!-- To add a body on the simulation, add the identifier and the body id att the same index.
  	You can find the names produced by gazebo inside the topic /gazebo/link_states on the name variable:
  	rostopic echo /gazebo/link_states/name -n 1 -->
	  <rosparam param="/body_array">[1,2,3,4,11,12,13,14]</rosparam>
	  <rosparam param="/body_names">["iris1::base_link","iris2::base_link","iris3::base_link","iris4::base_link","iris1::iris1/load_pose/odometry_sensorgt_link","iris2::iris2/load_pose/odometry_sensorgt_link","iris3::iris3/load_pose/odometry_sensorgt_link","iris4::iris4/load_pose/odometry_sensorgt_link"]</rosparam>

	 <!-- Launch the mocap node -->
	<node name="ros_mocap" pkg="mocap" type="ros_mocap_sim.py" respawn="true" output="screen">	
	</node>
	<!-- Launch the data recording node -->
        <node name="recorder" pkg="mocap" type="recorder.py" respawn="false" output="screen">
	</node>

  </group>

<!-- REAL QUAD -->
  <group unless="$(arg simulation)">
  	
  	  <!-- Set the frequency of the mocap publish rate -->
      <rosparam param="/frequency">20.</rosparam>

      <!-- Add a body tracked by mocap -->
      <!-- The identifier correspond to the position of the body that appears in the Qualisys list -->
      <!-- To add a body, add the Qualisys id and the name of the body (I am not sure that the names are still used) -->
	  <rosparam param="/body_array">[13,21,12,18,20]</rosparam>
	  <rosparam param="/body_names">["iris1","iris2","iris3","iris4","iris5"]</rosparam>

	  <!-- Launch the mocap node -->
	  <node name="ros_mocap" pkg="mocap" type="ros_mocap.py" respawn="true" output="screen">
	  </node>

        <!-- Launch the data recording node -->
        <node name="recorder" pkg="mocap" type="recorder.py" respawn="false" output="screen">
	</node>

  </group>

</launch> 
